packages/module/src/DataViewTh/DataViewTh.tsx

```diff
@@ -0,0 +1,266 @@
+    }
+
+    thRef.current?.style.setProperty('min-width', newSize + 'px');
+    currWidth = newSize;
+  };
+
+  const handleMouseup = (e: MouseEvent) => {
+    if (!isResizing.current) {
+      return;
+    }
+    // Reset the isResizing and dragOffset values to their initial states
+    isResizing.current = false;
+    dragOffset.current = 0;
+
+    // Call the onResize callback with the new width
+    onResize && onResize(e, currWidth);
+
+    // Handle scroll into view when column drag button is moved off screen
+    if (resizeButtonRef.current && !isInView.current) {
+      resizeButtonRef.current?.scrollIntoView({ behavior: 'smooth', block: 'center' });
+    }
+
+    document.removeEventListener('mousemove', callbackMouseMove);
+    document.removeEventListener('mouseup', callbackMouseUp);
+  };
+
+  const handleTouchEnd = (e: TouchEvent) => {
+    e.stopPropagation();
+    if (!isResizing) {
+      return;
+    }
+    // Reset the isResizing and dragOffset values to their initial states
+    isResizing.current = false;
+    dragOffset.current = 0;
+
+    // Call the onResize callback with the new width
+    onResize && onResize(e, currWidth);
+
+    document.removeEventListener('touchmove', callbackTouchMove);
+    document.removeEventListener('touchend', callbackTouchEnd);
+  };
+
+  const callbackMouseMove = useCallback(handleMouseMove, []);
+  const callbackTouchEnd = useCallback(handleTouchEnd, []);
+  const callbackTouchMove = useCallback(handleTouchMove, []);
+  const callbackMouseUp = useCallback(handleMouseup, []);
+
+  const handleKeys = (e: React.KeyboardEvent) => {
+    const key = e.key;
+
+    if (key !== 'Escape' && key !== 'Enter' && key !== 'ArrowLeft' && key !== 'ArrowRight') {
+      if (isResizing) {
+        e.preventDefault();
+      }
+      return;
+    }
+    e.preventDefault();
+
+    if (key === 'Escape' || key === 'Enter') {
+      onResize && onResize(e, currWidth);
+    }
+
+    const isRTL = getLanguageDirection(thRef.current as HTMLElement) === 'rtl';
+    const columnRect = thRef.current?.getBoundingClientRect();
+
+    let newSize = columnRect?.width || 0;
+    let delta = 0;
+    if (key === 'ArrowRight') {
+      if (isRTL) {
+        delta = -increment;
+      } else {
+        delta = increment;
+      }
+    } else if (key === 'ArrowLeft') {
+      if (isRTL) {
+        delta = increment;
+      } else {
+        delta = -increment;
+      }
+    }
+    newSize = newSize + delta;
+
+    thRef.current?.style.setProperty('min-width', newSize + 'px');
+    currWidth = newSize;
+  };
+
+  return (
+    <Th {...thProps} {...restProps} style={width > 0 ? { minWidth: width } : undefined} ref={thRef}>
+      {content}
+      {isResizable && (
+        <Button
+          ref={resizeButtonRef}
+          variant="plain"
+          onMouseDown={handleMousedown}
+          onKeyDown={handleKeys}
+          onTouchStart={handleTouchStart}
+          style={{ float: 'inline-end' }}
+        >
+          test
```

Comment:
Do wew have any icon to replace this text with at the moment? The end result would probably want to be this button's accessible name being something like `Reisze ${columnName}`.

Also,  maybe we should include some hidden text that can display the current column width (or a callback that can do so, for customization of the text language)? That should provide a means to announce that pressing the arrow keys on the button did something. Thinking as I type this, also wondering if we'd want to debounce the update to prevent a ton of announcements being thrown at users...

Reply:
@mcoker  is working on the icon + styling for the header cell. 

Will add some alt text / prop for alt text for the icon button.

Re: the hidden text, do you mean having a hidden div inside the th that is screen reader only and says something like "current width {width}"? Also how would we debounce that? Or should it be a callback function and I can add it to the example?